nix/desktop/default.nix

```diff
@@ -6,15 +6,15 @@ with stdenv;
 let
   targetLinux = {
     "linux" = true;
-    "" = isLinux;
+    "all" = isLinux;
   }.${target-os} or false;
   targetDarwin = {
     "macos" = true;
-    "" = isDarwin;
+    "all" = isDarwin;
   }.${target-os} or false;
   targetWindows = {
     "windows" = true;
-    "" = isLinux;
+    "all" = isLinux;
   }.${target-os} or false;
```

Comment:
Are we sure we don't want to keep the implicit `""` empty value to build everything?

Reply:
Yeah, there's no benefit in that. We control all values that are passed, so we can decide to be explicit. What we could add is logic to throw an exception if the value is unknown.

Reply:
Oh, that would be better!

Reply:
OK, added logic so that we get an error if an unknown value is passed in.